src/v0/endpoints/proposals/logic.rs

```diff
@@ -1,4 +1,22 @@
+use crate::db::{models::Proposal, schema::proposals::dsl::proposals};
 use crate::v0::context::{ChainData, SharedContext};
+use diesel::RunQueryDsl;
+
+pub async fn get_all_proposals(context: SharedContext) -> Vec<Proposal> {
+    let db_conn = context
+        .read()
+        .await
+        .db_connection_pool
+        .get()
+        .expect("Error connecting to database");
+    tokio::task::spawn_blocking(move || {
```

Comment:
I tried to create a function to abstract this call but couldn't make it work. I have a question in the Diesel gitter. Hopefully they answer and I can do it. For now it should be enough anyway.

Reply:
From the `diesel` gitter chat:

> You will need a lot more trait bounds to make something like that work. I normally do advice not to go down that route.

No harmful of having it duplicated.